runtime/contract_update_validation_test.go

```diff
@@ -1404,6 +1504,38 @@ func assertConformanceMismatchError(
 	)
 }
 
+func assertEnumCaseMismatchError(t *testing.T, err error, expectedEnumCase string, foundEnumCase string) {
+	require.Error(t, err)
+	require.IsType(t, &EnumCaseMismatchError{}, err)
+	enumMismatchError := err.(*EnumCaseMismatchError)
+
+	assert.Equal(
+		t,
+		fmt.Sprintf(
+			"mismatching enum case: expected `%s`, found `%s`",
+			expectedEnumCase,
+			foundEnumCase,
+		),
+		enumMismatchError.Error(),
```

Comment:
Maybe instead of comparing error messages literally, just check declName, expectedEnumCase, and foundEnumCase of the error directly, so changing the error message doesn't break the test

Reply:
Also wanted to assert the error message, to make sure they are consistent, and the to-string works with nested errors, etc.